File: 백준/Silver/2108. 통계학/통계학.py
import sys
from collections import Counter

N = int(sys.stdin.readline().strip())

nums = []
for _ in range(N):
  nums.append(int(sys.stdin.readline().strip()))

nums.sort()
avg = sum(nums) / len(nums)

def my_round(val):
  if val - int(val) >= 0.5:
    return int(val)+1
  else:
    return int(val)

# 평균값
print(round(sum(nums) / len(nums)))
# print(my_round(sum(nums) / len(nums)))


# 중앙값
print(nums[len(nums) // 2])

# 최빈값
cnt = Counter(nums)
# 상위 두 개만 보기
if len(cnt) != 1:
  mode = cnt.most_common(2)
  if mode[0][1] == mode[1][1]:
    print(mode[1][0])
  else:
    print(mode[0][0])
else:
  mode = cnt.most_common(1)
  print(mode[0][0])

# 범위
print(nums[len(nums)-1]-nums[0])

File: 백준/Silver/2108. 통계학/통계학.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());
        int[] nums = new int[N];

        int sum = 0;
        for (int i = 0; i < N; i++){
            nums[i] = Integer.parseInt(br.readLine());
            sum += nums[i];
        }
        Arrays.sort(nums);

        int count = 0;
        int max = -1;
        int mod = nums[0];
        boolean flag = false;
        for(int i = 0; i < N - 1; i++) {
            if(nums[i] == nums[i + 1])      count++;
            else                            count = 0;

            if(max < count) {
                max = count;
                mod = nums[i];
                flag = true;
            }else if(max == count && flag) {
                mod = nums[i];
                flag = false;
            }
        }

        System.out.println(Math.round((double) sum / N));
        System.out.println(nums[N / 2]);
        System.out.println(mod);
        System.out.println(nums[N - 1] - nums[0]);
    }
}